File: luogu/problems/P1161-Turning-On-the-Lights/solution.py
# P1161 开灯
# Author: Dhgaj
# Date: 2026-05-22


def main():
    n = int(input())

    # 存储当前亮着的灯
    lights = set()

    for _ in range(n):

        # 读取输入
        ai, ti = input().split()

        ai = float(ai)
        ti = int(ti)

        # 模拟操作
        for j in range(1, ti + 1):

            # 灯编号
            num = int(ai * j)

            # 翻转状态
            if num in lights:
                lights.remove(num)
            else:
                lights.add(num)

    # 最终只剩一个
    print(min(lights))


if __name__ == "__main__":
    main()

File: luogu/problems/P1161-Turning-On-the-Lights/solution.cpp
// P1161 开灯
// Author: Dhgaj
// Date: 2026-05-22

#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    // 核心逻辑
    int solve()
    {
        // 操作组数
        int n;

        cin >> n;

        // 灯状态
        vector<bool> light(2000001, false);

        // 输入数据
        double a;
        int t;

        // 处理每组操作
        for (int i = 0; i < n; i++)
        {
            // 读取输入
            cin >> a >> t;

            // 模拟翻转
            for (int j = 1; j <= t; j++)
            {
                // 灯编号
                int num = (int)(a * j);

                // 异或翻转
                light[num] = !light[num];
            }
        }

        // 找最终亮着的灯
        for (int i = 0; i <= 2000000; i++)
        {
            if (light[i])
            {
                return i;
            }
        }

        return 0;
    }
};

int main()
{
    // 实例化对象
    Solution solution;

    // 调用核心逻辑
    int answer = solution.solve();
    cout << answer;

    return 0;
}